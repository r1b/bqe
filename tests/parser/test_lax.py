import pytest


@pytest.mark.parametrize(
    "sql",
    (
        pytest.param("a and b and c", id="lax_expr_and"),
        pytest.param("case when a and b then c else d end", id="lax_expr_case"),
        pytest.param("|> SELECT a, b, c", id="lax_expr_pipe_select"),
        pytest.param("|> AGGREGATE GROUP BY a, b, ROLLUP(c,d)", id="lax_expr_pipe_agg"),
    ),
)
def test_lax_ok(sql, assert_parse_tree):
    assert_parse_tree(sql, lax=True)
