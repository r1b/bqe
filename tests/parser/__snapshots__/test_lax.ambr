# serializer version: 1
# name: test_lax_ok[lax_expr_and][sql]
  'a and b and c'
# ---
# name: test_lax_ok[lax_expr_and][tree]
  '''
  and_expr
    and_expr
      ident	a
      ident	b
    ident	c
  
  '''
# ---
# name: test_lax_ok[lax_expr_case][sql]
  'case when a and b then c else d end'
# ---
# name: test_lax_ok[lax_expr_case][tree]
  '''
  case_expr
    when_then_expr
      and_expr
        ident	a
        ident	b
      ident	c
    else_expr
      ident	d
  
  '''
# ---
# name: test_lax_ok[lax_expr_pipe_agg][sql]
  '|> AGGREGATE GROUP BY a, b, ROLLUP(c,d)'
# ---
# name: test_lax_ok[lax_expr_pipe_agg][tree]
  '''
  pipe_agg
    group_by
      grouping_items
        grouping_item
          ident	a
        grouping_item
          ident	b
        grouping_item
          rollup_list
            rollup_item
              ident	c
            rollup_item
              ident	d
  
  '''
# ---
# name: test_lax_ok[lax_expr_pipe_select][sql]
  '|> SELECT a, b, c'
# ---
# name: test_lax_ok[lax_expr_pipe_select][tree]
  '''
  pipe_select
    select_list
      select_expr
        ident	a
      select_expr
        ident	b
      select_expr
        ident	c
  
  '''
# ---
